packages/backend-platform/src/WebsocketService.ts

```diff
@@ -0,0 +1,1348 @@
+      reconnectDelay: options.reconnectDelay ?? 500,
+      maxReconnectDelay: options.maxReconnectDelay ?? 5000,
+      requestTimeout: options.requestTimeout ?? 30000,
+    };
+
+    // Register action handlers
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:init`,
+      this.init.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:connect`,
+      this.connect.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:disconnect`,
+      this.disconnect.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:sendMessage`,
+      this.sendMessage.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:sendRequest`,
+      this.sendRequest.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:getConnectionInfo`,
+      this.getConnectionInfo.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:getSubscriptionByChannel`,
+      this.getSubscriptionByChannel.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:isChannelSubscribed`,
+      this.isChannelSubscribed.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:addChannelCallback`,
+      this.addChannelCallback.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:removeChannelCallback`,
+      this.removeChannelCallback.bind(this),
+    );
+
+    this.#messenger.registerActionHandler(
+      `BackendWebSocketService:getChannelCallbacks`,
+      this.getChannelCallbacks.bind(this),
+    );
+
+    this.init().catch((error) => {
+      console.error('WebSocket service initialization failed:', error);
+    });
+  }
+
+  /**
+   * Initializes and connects the WebSocket service
+   *
+   * @returns Promise that resolves when initialization is complete
+   */
+  async init(): Promise<void> {
+    try {
+      await this.connect();
+    } catch (error) {
+      const errorMessage =
+        error instanceof Error ? error.message : 'Unknown initialization error';
+
+      throw new Error(
+        `WebSocket service initialization failed: ${errorMessage}`,
+      );
+    }
+  }
+
+  /**
+   * Establishes WebSocket connection
+   *
+   * @returns Promise that resolves when connection is established
+   */
+  async connect(): Promise<void> {
+    // If already connected, return immediately
+    if (this.#state === WebSocketState.CONNECTED) {
+      return;
+    }
+
+    // If already connecting, wait for the existing connection attempt to complete
+    if (this.#state === WebSocketState.CONNECTING && this.#connectionPromise) {
+      await this.#connectionPromise;
+      return;
+    }
+
+    console.log(`🔄 Starting connection attempt to ${this.#options.url}`);
```

Comment:
It looks like you are using various log levels to print messages, so what are your thoughts on using `loglevel` instead of `console.*`? This is a dependency we use in other `core` packages, and it allows log messages to be controlled at the build level (so we can show `console.log` errors in dev but only `console.info` errors in prod, for instance).